iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HiveIcebergRESTCatalogClientAdapter.java

```diff
@@ -0,0 +1,442 @@
+import org.apache.hadoop.hive.metastore.api.SQLUniqueConstraint;
+import org.apache.hadoop.hive.metastore.api.SerDeInfo;
+import org.apache.hadoop.hive.metastore.api.ShowLocksResponse;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.api.UnknownDBException;
+import org.apache.hadoop.hive.metastore.api.WMFullResourcePlan;
+import org.apache.hadoop.hive.serde.serdeConstants;
+import org.apache.iceberg.BaseTable;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SessionCatalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.rest.RESTCatalog;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HiveIcebergRESTCatalogClientAdapter implements IMetaStoreClient {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HiveIcebergRESTCatalogClientAdapter.class);
+  public static final String NAMESPACE_SEPARATOR = ".";
+  public static final String NAME = "name";
+  public static final String LOCATION = "location";
+  public static final String CATALOG_NAME = "iceberg.catalog";
+  public static final String DB_OWNER = "owner";
+  public static final String DB_OWNER_TYPE = "ownerType";
+  public static final String DEFAULT_INPUT_FORMAT_CLASS = "org.apache.iceberg.mr.hive.HiveIcebergInputFormat";
+  public static final String DEFAULT_OUTPUT_FORMAT_CLASS
+      = "org.apache.iceberg.mr.hive.HiveIcebergOutputFormat";
+  public static final String DEFAULT_SERDE_CLASS = "org.apache.iceberg.mr.hive.HiveIcebergSerDe";
+  public static final String CATALOG_CONFIG_PREFIX = "iceberg.catalog.";
+  private final Configuration conf;
+  private RESTCatalog restCatalog;
+  private final HiveMetaHookLoader hookLoader;
+
+  private final long maxHiveTablePropertySize;
+
+  public HiveIcebergRESTCatalogClientAdapter(Configuration conf, HiveMetaHookLoader hookLoader) {
+    this.conf = conf;
+    this.hookLoader = hookLoader;
+    this.maxHiveTablePropertySize = conf.getLong(HiveOperationsBase.HIVE_TABLE_PROPERTY_MAX_SIZE,
+          HiveOperationsBase.HIVE_TABLE_PROPERTY_MAX_SIZE_DEFAULT);
+  }
+
+  @Override
+  public void reconnect() throws MetaException {
+    SessionCatalog.SessionContext context = SessionCatalog.SessionContext.createEmpty();
+    String catalogName = conf.get(CATALOG_NAME);
+    Map<String, String> properties = getCatalogPropertiesFromConf(conf, catalogName);
+    restCatalog = (RESTCatalog) CatalogUtil.buildIcebergCatalog(catalogName, properties, conf);
```

Comment:
why not explicitly create RestCatalog? example:
````
  private RESTCatalog initCatalog(String catalogName) throws Exception {
    URI catalogUri = URI.create("http://localhost:" + catalogPort + "/" + catalogPath);
    RESTCatalog catalog = new RESTCatalog();

    Map<String, String> properties = Maps.newHashMap();
    properties.put(CatalogProperties.URI, catalogUri.toString());
    if (authType == ServletSecurity.AuthType.JWT) {
      properties.put("token", generateJWT());
    }
    catalog.initialize(catalogName, properties);
    return catalog;
  }
````

Reply:
Fixed.